Use on_repair when repairing tube

S-S-X opened the pull request:
Fixes #36 

I've not tested, not even launched mt. But GH web editor is way better nowdays 😄

wsor4035 commented:
mods should use the _ prefix per lua_api.txt

OgelGames commented:
Yeah, no one uses the prefix :P I would suggest putting it inside the `tube` table though, to keep all the pipeworks stuff in one place.

Side note: `is_teleport_tube` is also defined outside of the `tube` table, but I noticed it's only used in one place:
https://github.com/mt-mods/pipeworks/blob/37eef736950f038e54fd74d11a5d905fb43abdad/teleport_tube.lua#L124
Might be good to change this to check the name instead, so that value can be removed. (in another PR)

S-S-X commented on teleport_tube.lua:
```diff
@@ -259,6 +259,18 @@ pipeworks.register_tube("pipeworks:teleport_tube", {
 		on_destruct = function(pos)
 			remove_tube(pos)
+		end,
+		on_repair = function(pos, node)
```
64fb579 moved to nodedef.tube.on_repair

S-S-X commented:
Switched to nodedef.tube.on_repair, also added extra guard against possible unknown node crash (very unlikely but possible and cheap check).
I've still not tested or launched server with these changes.

OgelGames commented:
> Not sure if swap_node can tolerate unknown...

Yeah it can't:
```
AsyncErr: Lua: Runtime error from mod 'pipeworks' in callback ScriptApiNode::node_on_punch(): "nothing:nothing" is not a registered node!
stack traceback:
	[C]: in function 'swap_node'
	C:\Minetest\client\bin\..\mods\pipeworks/routing_tubes.lua:87: in function <C:\Minetest\client\bin\..\mods\pipeworks/routing_tubes.lua:63>
```
Seems to work correctly otherwise, at least from my limited testing.

S-S-X commented:
Now unknown node situation is checked for all paths instead of just for new functionality, situation is logged and also message sent to player telling that tube cannot be repaired.
